iothub/device/samples/getting started/MessageReceiveSample/MessageReceiveSample.cs

```diff
@@ -104,13 +112,15 @@ private async Task ReceiveC2dMessagesPollingAndCompleteAsync(CancellationToken c
 
         private async Task OnC2dMessageReceivedAsync(Message receivedMessage, object _)
         {
+            await _processMessageSemaphore.WaitAsync(_cts.Token).ConfigureAwait(false);
             Console.WriteLine($"{DateTime.Now}> C2D message callback - message received with Id={receivedMessage.MessageId}.");
             PrintMessage(receivedMessage);
 
             await _deviceClient.CompleteAsync(receivedMessage);
             Console.WriteLine($"{DateTime.Now}> Completed C2D message with Id={receivedMessage.MessageId}.");
 
             receivedMessage.Dispose();
+            _processMessageSemaphore.Release();
```

Comment:
This should be in a try/finally so it is always released, even if the code above crashes.